src/systems/fem_system.C

```diff
@@ -74,7 +75,7 @@ void assemble_unconstrained_element_system(const FEMSystem & _sys,
 
   // We need jacobians to do heterogeneous residual constraints
   const bool need_jacobian =
-    _get_jacobian || _constrain_heterogeneously;
+    (_get_jacobian || _constrain_heterogeneously) && (!_no_constraints);
```

Comment:
This can't be right, can it?  If the user wants an unconstrained Jacobian, then _get_jacobian is true, _no_constraints is true, and need_jacobian will be *false*?